pkg/apis/etcd/v1beta2/backup_types.go

```diff
@@ -131,3 +138,19 @@ type ABSBackupSource struct {
 	// The name of the secret object that stores the Azure storage credential
 	ABSSecret string `json:"absSecret"`
 }
+
+// GCSBackupSource provides the spec how to store backups on GCS.
+type GCSBackupSource struct {
+	// Path is the full GCS path where the backup is saved.
+	// The format of the path must be: "<gcs-bucket-name>/<path-to-backup-file>"
+	// e.g: "mygcsbucket/etcd.backup"
+	Path string `json:"path"`
+
+	// The name of the secret object that stores the Google storage credential
+	// containing at most ONE of the following:
+	// An access token with file name of 'access-token'.
+	// JSON credentials with file name of 'credentials.json'.
+	//
+	// If omitted, client will use the default application credentials.
+	GCPSecret string `json:"gcpSecret,omitempty"`
```

Comment:
@cpick

Reply:
The current "gcpSecret" implementation follows the lead of the existing backup config's 
[clientTLSSecret](https://github.com/coreos/etcd-operator/blob/master/pkg/apis/etcd/v1beta2/backup_types.go#L75) and backup backend implementations' [awsSecret](https://github.com/coreos/etcd-operator/blob/master/pkg/apis/etcd/v1beta2/backup_types.go#L117) and [absSecret](https://github.com/coreos/etcd-operator/blob/master/pkg/apis/etcd/v1beta2/backup_types.go#L132).

I agree it would be nice if there was better namespace support across the operator, but I think it would be reasonable to add this GCS backend with an implementation that matches the existing backends and then possibly improve namespace support in a different pull request that addresses the issue more comprehensively.

Reply:
Noted, thanks! I can contribute the follow-up PR. :)

Reply:
@cpick 
After going through the codebase more thoroughly, I think if we change the `gcpSecret` field to `SecretReference` type, it is better to modify corresponding AWS and ABS APIs too for API consistency and clarity.

After careful thought, I think I will leave the APIs as they are (i.e., `gcpSecret` field is still a string type). We can support `SecretReference` later if there is enough community support.